File: src/printable.py
#! /usr/bin/env python3

# printable.py - print a table of printable ASCII characters in base 2, 8, 10, and 16

for i in range(32, 127):
  print(f"{i:07b} | {i:03o} | {i:03d} | {i:2x} | {chr(i):1s}")

File: src/printable.c
// printable.c - print a table of printable ASCII characters in bases 16 and 10

#include <locale.h>
#include <stdio.h>
#include <wchar.h>

// Todo: Move this up when Clang defines the macro in the environment
#ifndef __STDC_ISO_10646__
#error
#endif

int main(void) {
  if (!setlocale(LC_ALL, ""))
    return 1;

  for (wint_t code = 32; code <= 126; ++code)
    printf("%lc | %x | %.3d\n", code, code, code);
}
